### PR TITLE
feat(j-s): Get police case info

### PR DESCRIPTION
**Pull request by unakb**
[Búa til nýtt function í police service sem skilar upplýsingum um mál](https://app.asana.com/0/1199153462262248/1204823390153048/f)

## What

Retrieve relevant information from police cases

## Why

So that we can use information that's already been saved to the police system instead of users having to input it again

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/police/models/policeCaseInfo.model.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { ApiProperty } from '@nestjs/swagger'
+
+export class PoliceCaseInfo {
+  @ApiProperty()
+  caseNumber!: string
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/police/models/policeCaseInfo.model.ts:5]: Please rename to `policeCaseNumber` to be consistant with other models.

File: apps/judicial-system/backend/src/app/modules/police/models/policeCaseInfo.model.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  @ApiProperty()
+  crimeScene?: string
+  @ApiProperty()
+  crimeDate?: Date
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/police/models/policeCaseInfo.model.ts:9]: Please see comments to the corresponding model in the api project.

File: apps/judicial-system/backend/src/app/modules/police/police.service.ts [hunk 3]
```diff
@@ -247,6 +248,94 @@ export class PoliceService {
...
+    const promise = this.fetchPoliceDocumentApi(
+      `${this.xRoadPath}/V2/GetDocumentListById/${caseId}`,
+    )
+
+    return promise
+      .then(async (res: Response) => {
+        if (res.ok) {
+          const response = await res
+            .json()
+            .then((result) =>
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/police/police.service.ts:264]: Please consider removing this conditional second JSON parse. The api has been fixed, so we no longer need to handle this here.

**Review by gudjong — APPROVED** [review]
Some naming considerations

File: apps/judicial-system/api/src/app/data-sources/backend.ts [hunk 4]
```diff
@@ -46,6 +46,7 @@ import {
   IndictmentCount,
   UpdateIndictmentCountInput,
 } from '../modules/indictment-count'
+import { PoliceCaseInfo } from '../modules/police/models/policeCaseInfo.model'
```
[the remark this answers is not in the document]
Comment (unakb) [reply apps/judicial-system/api/src/app/data-sources/backend.ts:49]: This is the only model being imported from this area right now and there's no existing index file that makes sense to put this in, so I'm not sure I see the benefit. The other types are being imported from the types library which I decided not to do for PoliceCaseInfo as I didn't add it to types and want to use the generated schema instead.

[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/api/src/app/data-sources/backend.ts]: The idea in the api and backend projects is to have a single index file at the top of each module. Everything exported from the module is exported from the index file. No index files are used within the modules themselves. This was done to make it a bit clearer which types/classes/enums/etc are need outside each module.

**Pull request merged**